### PR TITLE
docs: replaced manual up next blocks with doc tag in workspaces.md

### PR DESCRIPTION
**Pull request by rzubov**
**What has been done**
Updated workspace.md - changed Up Next block, replaced manually defined pages with doc tag

Related PR:

### CODE REVIEW

**Review by antonkor — APPROVED** [review]

**Comment by rzubov** [comment]
@mafredri I was looking on <children> tag while implementing this
https://github.com/coder/docs/blob/main/admin/index.md?plain=1#L20

**Comment by antonkor** [comment]
> This'll look a bit weird for anyone viewing the docs on GitHub, the heading will be visible but `<doc>` doesn't show up. Do we want to maintain compatibility or guide people towards not using the docs in the repo?

Hmm. I didn't foresee this issue when planning out this task. Originally I was trying to reduce the markup so the contributors had less to write.

I think we should definitely push users to our site's docs for marketing/website goals wherever we can on outside platforms

I might have a solution, I'll try submitting a proposal tonight.

**Comment by antonkor** [comment]
What do you guys think about this workaround?

Allow contributors to use markdown like they normaly would so github gets the markdown as intended:

Screen Shot 2022-07-20 at 12 29 41 AM

then in the markdown parser on our site, we use regex to select ## Up next or ## Next steps and convert any markdown links in that ## section into those cards. We'll have to do some testing to make sure performance isn't affected.

this way contributors have less to write when linking the next steps but we get those fancy cards.

ofc the other option is letting the up next cards be HTML. not pretty but the simplest solution here. 

let me know how you what you think.

Also, what are your thoughts on adding a cta to the bottom of all the docs on github to get people into our site's docs? If we use the markdown parser solution we can strip it out when the docs get rendered on our site, this way the cta will only appear on github docs. i have a few text variants in figma.

Screen Shot 2022-07-20 at 12 50 33 AM

figma for ref: https://www.figma.com/file/LYL5VURCZ5lMgtUDiuEvER/Coder-V2-(In-Process)?node-id=563%3A2044

**Comment by ammario** [comment]
I think that solution is good.


What do you mean by the `docs site` thing? Do you mean that every markdown doc should have that in the footer?

**Comment by antonkor** [comment]
> What do you mean by the `docs site` thing? Do you mean that every markdown doc should have that in the footer?

Yeah, i think it's gonna be good to try to get that traffic back to our docs through a cta/link on every docs .md file. We can have a global link for each of the 3 docs. 

It will not only direct users from GitHub back to our domain but also the people who use docs in their IDEs. the more returning users to our site the better I think. 

the only downside is it's a bit repetitive for this use (and to maintain, search and replace ftw ;) but it should be worth it for all the extra traffic we can get back. another option is to use it sparingly on the top-performing doc pages (based on GA).

I'll assign roman to do the regex implementation of the cards. and I can analyze GA what are the top visiting pages on each of the docs to add that CTA, if you think we should go the sparingly route vs all pages.

**Comment by ammario** [comment]
I think something like that would be good for the README doc, but it would be too repetitive and easy to forget for the rest.

**Comment by ammario** [comment]
Just waiting on Bruno to approve https://github.com/coder/coder.com/pull/84 before merging this in.

**Comment by BrunoQuaresma** [comment]
> Just waiting on @BrunoQuaresma to approve [coder/coder.com#84](https://github.com/coder/coder.com/pull/84) before merging this in.

It is approved!

**Review by antonkor — APPROVED** [review]

**Comment by ammario** [comment]
Screen Shot 2022-08-01 at 4 03 59 PM

Hmm.. @antonkor I'm not seeing these blocks format correctly. I'm running `yarn dev` on coder.com's master.

**Comment by rzubov** [comment]
@ammario  PR to conert links to markup is not merged yet
https://github.com/coder/coder.com/pull/101

File: docs/templates.md
Comment (ammario) [inline docs/templates.md:240]: Why use Next Steps here but Up next in the README.md?

**Comment by jsjoeio** [comment]
@rzubov anything I can do to help move this forward?

File: docs/templates.md
Comment (antonkor) [reply docs/templates.md:240]: @ammario the text can be interchanged; In most places, Next Steps was used, in a few spots Up next was used. not sure why there was a slight difference in terminology so i left it as is and had roman extend the regex to include both wordings and thought the docs authors would sort it out. 

I can change all the text to read Next Steps, just let me know :)

(sorry i missed this til now)

**Comment by antonkor** [comment]
> @rzubov anything I can do to help move this forward?

I missed this issue in my inbox til now. Lets see how Ammar responds. 

@jsjoeio Is there anything I need todo to resolve the "release / linux-windows (push)" and "coder / test/go (windows-2022) (push)"  checks? Or is that stuff we can overlook?

**Comment by jsjoeio** [comment]
> @jsjoeio Is there anything I need todo to resolve the "release / linux-windows (push)" and "coder / test/go (windows-2022) (push)" checks? Or is that stuff we can overlook?

Try resolving your conflicts and rebasing on top of `main`, pushing and seeing if that fixes it 🤞🏼 Hoping just a flake.

**Comment by rzubov** [comment]
@jsjoeio  Conflict are resolved

**Review by jsjoeio — APPROVED** [review]
This looks good to me! Thank you so much!

If @ammario requests changes after we merge, it's an easy fix.

**Pull request merged**
